tasks.gradle

```diff
@@ -84,38 +84,127 @@ processPropertiesFiles(sourceDir)
-            if (!project.findProperty("prompt")) {
-                if (errorMsg != null) {
-                    project.logger.error(errorMsg)
-                } else {
-                    project.logger.lifecycle("🤖 Please provide the parameter: prompt (Ex. -Pprompt=\"How are you?\")")
-                }
-                return
+        def reader = new BufferedReader(new InputStreamReader(System.in))
+        def yellowText = "\u001B[33;1m"
+        def greenText = "\u001B[32;1m"
+        def blueText = "\u001B[34;1m"
+        def redText = "\u001B[31;1m"
+        def resetText = "\u001B[0m"
+
+        def etendoHost = project.findProperty('host')
+        def username = project.findProperty('username')
+        def password = project.findProperty('password')
+        def app_id = project.findProperty("app_id")
+        String question = project.findProperty("question")
+        if (project.hasProperty("single")) {
+            project.logger.lifecycle("${yellowText}🤖 Single mode enabled. Only one question will be asked.${resetText}")
+        }
+
+        if (username == null || username == "") {
+            project.logger.lifecycle("${blueText}👮 Please provide the username:${resetText}")
+            username = reader.readLine()?.trim()
+        }
+        if (password == null || password == "") {
+            project.logger.lifecycle("${blueText}👮 Please provide the password:${resetText}")
+            password = reader.readLine()?.trim()
+            exec {
+                commandLine 'clear'
             }
-            sharedPrompt = project.findProperty("prompt")
         }
-        project.logger.lifecycle("🤖 I'm working on your request: $sharedPrompt")
-        String translationURL = "http://0.0.0.0:${project.ext.get(COPILOT_PORT_PROPERTY)}/question"
+        if (!etendoHost) {
+            project.logger.lifecycle("${redText}🤖 Host not provided, using default (http://localhost:8080/etendo).${resetText}")
+            etendoHost="http://localhost:8080/etendo"
+        }
+        if (app_id == null || app_id == "") {
+            project.logger.lifecycle("${redText}🤖 Please provide the parameter: app_id (Ex. -Papp_id=\'A1B2C3D4E5F6G7H8\').${resetText}")
+            return
+        }
 
-        HttpClient client = HttpClient.newHttpClient()
-        HttpRequest request = HttpRequest.newBuilder()
-                .uri(URI.create(translationURL))
+        String login = JsonOutput.toJson([
+                "username": username,
+                "password": password
+        ])
+
+        String loginURL = "${etendoHost}/sws/login"
+
+        HttpClient hClient = HttpClient.newHttpClient()
+        HttpRequest hRequest = HttpRequest.newBuilder()
+                .uri(URI.create(loginURL))
                 .header("Content-Type", "application/json")
-                .POST(HttpRequest.BodyPublishers.ofString("{\"question\":\"${sharedPrompt}\"}"))
+                .method("GET", HttpRequest.BodyPublishers.ofString(login))
```

Comment:
To follow good practice, a GET should never have a body, POST is correct. Also, it is important that the role with which it is to be executed, for simplicity, one option is to add the payload to role 0. This would have the effect that it can only be used by a ‘System Admin’, which I think is normal for this mode of the Copilot system.